static/js/protein/chart.ts

```diff
@@ -21,18 +21,80 @@ const XLINKNS = "http://www.w3.org/1999/xlink";
 
 const HEIGHT = 224;
 const WIDTH = 500;
+const MARGIN = { top: 70, right: 50, bottom: 10, left: 50 };
 
+// interface for protein page datatypes which return number values
+export interface ProteinPropDataNumType {
+  name: string;
+  value: number;
+}
+
+// interface for protein page datatypes which return string values
+export interface ProteinPropDataStrType {
+  name: string;
+  value: string;
+}
 /**
  * Draw bar chart for tissue score.
  */
 export function drawTissueScoreChart(
   id: string,
   data: { name: string; value: string }[]
 ): void {
-  // TODO: convert data to tmpData format.
-  const tmpData = [{ x1: 20, x2: 60, y1: 30, y2: 50 }];
+  /*
+  Convert the tissue expression level to numerical values 
+  * @param val value of the data array 
+  */
+  function convertData(val) {
+    if (val == "ProteinExpressionNotDetected") {
+      return 0;
+    } else if (val == "ProteinExpressionLow") {
+      return 1;
+    } else if (val == "ProteinExpressionMedium") {
+      return 2;
+    } else if (val == "ProteinExpressionHigh") {
+      return 3;
+    }
+  }
+
+  var reformattedData = {} as ProteinPropDataNumType[];
+  reformattedData = data.map((item) => {
+    return {
+    name: item.name,
+    value: convertData(item.value),
+    }
+  });
+
+  const arr_name = reformattedData.map((x) => {
+    return x.name;
+  });
+
+  const max_height = 400;
+  const max_width = 460;
+  const height = max_height - MARGIN.top - MARGIN.bottom;
+  const width = max_width - MARGIN.left - MARGIN.right;
+  const x = d3
+    .scaleLinear()
+    .domain([0, d3.max(reformattedData, (d) => d.value)])
+    .range([MARGIN.left, width - MARGIN.right]);
+  const y = d3
+    .scaleBand()
+    .domain(arr_name)
+    .rangeRound([MARGIN.top, height - MARGIN.bottom])
+    .padding(0.1);
+  function formatTick(d) {
+    if (d == 0) {
```

Comment:
would be nice if some of these were defined in constant dictionaries at the top of the file, so updates in the future to colors, etc, would be easier to make.
```js
const TISSUE_SCORE_TO_LABEL = {
   0: "NotDetected",
   1: "Low",
   ...
}
...
.tickFormat((d) => TISSUE_SCORE_TO_LABEL[d])
```